platform/engineering/request-for-comment/2022-05-05-mobile-route-naming-conventions.md

```diff
@@ -0,0 +1,201 @@
+# RFC: Mobile Routes naming conventions
+
+- Date: 2022-05-05
+- Related Issue: https://app.zenhub.com/workspaces/va-mobile---backend-61d75d03a7505500147ec7f6/issues/department-of-veterans-affairs/va-mobile-app/2937
+
+## Background
+Mobile routes currently have some inconsistencies in their naming conventions, such as:
+- Sometimes using snake_case over kebab-case (maintenance_windows, message_drafts)
+- :ids are after the resource action rather than before (canceling appointments)
+- Not nesting under parent products (the 'health' and 'benefits' app)
+
+## Motivation
+Keeping naming consistent will make using the Mobile API much easier to use and is generally a good practice for maintainability. 
+
+## Design
+Version all endpoint that do not meet the naming convention as follows:
+```
+namespace :health do
+  get '/appointment-requests/:appointment_request_id/messages', to: 'appointment_request_messages#index'
+  get '/appointments', to: 'appointments#index'
+  put '/appointments/:id/cancel', to: 'appointments#cancel'
+  get '/appointments/community-care/:service_type/eligibility', to: 'community_care_eligibility#show'
+  get '/appointments/va/eligibility', to: 'veterans_affairs_eligibility#show'
```

Comment:
`/appointments/eligibility/va`, `/appointments/eligibility/facility` ?

Reply:
There's community care too 😁 and although community-care could be a parent I think it has to do more with 'can' rather than 'where' it could also fall under eligibility `/appointments/eligibility/community-care` ?